.github/workflows/build_nuget.yml

```diff
@@ -0,0 +1,110 @@
+          boost-thread
+          boost-program-options
+          boost-test
+          boost-align
+          boost-math
+          boost-uuid
+          boost-system
+          boost-date-time
+          boost-regex
+          boost-asio
+          boost-interprocess
+      - name: Configure
+        run: >
+          cmake -S . -B build
+          -G "Visual Studio 16 2019" -A ${{ matrix.arch_config.generator_arch }}
+          -T ${{ matrix.toolset }}
+          -DRL_BUILD_NUGET=On
+          -DRL_NUGET_PACKAGE_NAME=RLClientLibNativeStatic
+          -DRL_NUGET_PACKAGE_VERSION="0.1.0"
+          -DNATIVE_NUGET_PLATFORM_TAG=${{ matrix.arch_config.arch }}
+          -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake
+          -DVCPKG_TARGET_TRIPLET=${{ matrix.arch_config.arch }}-windows
+          -Drlclientlib_BUILD_DOTNET=Off
+      - name: Compile - debug build
+        run: cmake --build ./build --config Debug -v
+      - name: Create Nuget package - debug build
+        # this copies over files to nuget_staging and creates a package with only debug libraries
+        run: cmake --install ./build --config Debug --prefix ./nuget_staging -v
+      - name: Clean
+        run: cmake --build ./build --target clean
+      - name: Compile - release build
+        run: cmake --build ./build --config Release -v
+      - name: Create Nuget package - release build
+        # this overwrites the previous package with one containing both debug and release libraries
+        run: cmake --install ./build --config Release --prefix ./nuget_staging -v
+      - name: Get package name
+        id: nuget_name
+        run: |
+          cd nuget_staging
+          ls
+          $NugetFileName = Get-ChildItem *.nupkg -name
+          echo "::set-output name=NugetFileName::$NugetFileName"
+      - name: Upload package
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{ steps.nuget_name.outputs.NugetFileName }}
+          path: nuget_staging/${{ steps.nuget_name.outputs.NugetFileName }}
+
+  test_nuget_windows:
+    needs: [build_nuget_windows]
+    name: nuget-test.${{ matrix.toolset }}
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: ["windows-2019"]
+        toolset: ["v141", "v142"]
+        nuget_package_version: ["0.1.0"]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+      - uses: ilammy/msvc-dev-cmd@v1
+      - uses: actions/download-artifact@v1
+        with:
+          name: RLClientLibNativeStatic-${{matrix.toolset}}-x64.${{matrix.nuget_package_version}}.nupkg
+          path: downloaded_nugets
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.1
+      - name: Build test
+        run: |
+          ls downloaded_nugets
+          .\nuget\nuget.exe install -Source ${{ github.workspace }}\downloaded_nugets -OutputDirectory ${{ github.workspace }}\nuget\test\packages\ RLClientLibNativeStatic-${{ matrix.toolset }}-x64
```

Comment:
You may want to create some variable to ensure that this nuget.exe is the same as the one found in the nuget cmake

Reply:
That seems rather cumbersome... the build and test run on separate VM instances. I'll try using system nuget.exe for the test.

Anyways, we don't expect the end user to have the exact same nuget.exe as we're building the package with, so I don't see a need to use the exact same one for running the test.